### DISCUSSION TITLE
Tasmota integration

### DISCUSSION THREAD

**Discussion by michael-bartmann** [universam1/iSpindel#556] [Ideas]
Recently I got my own iSpindel. The firmware worked right from the start. I am impressed. But there are certain features, which I know from dozens of tasmota-driven devices in my home network. E.g. MQTT over TLS (which all of my other devices use). And being able to update parameters (and firmware!) remotely.

Back in the day tasmota (started 2016) had no support for deep sleep, so it was not suitable for battery operation. But it seems this has changed with recent versions. For my personal use, which is local MQTT, tasmota might even suffice.

I understand that a lot of work has been but into iSpindel firmware. And many of the features are not available with tasmota as of now (Ubidots BeerBlox, CraftBeerPi integrations). But then again there is a big overlap, where tasmota does a great job (e.g. OTA firmware update, MQTT, TLS, local rule system, timers). Admittedly some of tasmota features would not be applicable to iSpindel (no actors).

If I use tasmota with deep sleep and get support for the iSpindel accelerometer, this might already be enough for my application. I would have to check if the current deep sleep of tasmota is good enough to allow for a decent battery life.

But maybe there are better ideas how to integrate more deeply, using more of the code of iSpindel, sharing the best of two worlds. My impression is that tasmota has become a kind of IOT operating system for devices like wemos.

I hope this idea does not sound offensive, as it questions part of the code of iSpindle firmware which would become obsolete. Or are there known counterindications already? I did not find any discussion of this in this forum (or whole google-space).

What are your ideas about this? 

Regards,
Michael

**Comment by thegreatgunbantoad** [comment]
Tasmota looks interesting, but the documentation is not very helpful.  What can it do?

**Reply by michael-bartmann** [reply]
It is a firmware normally used for any iot devices around esp8266 or esp32 microcontrollers. The "OTA" in the name stands for the integrated "over the air" firmware update feature. It has modules for support of a multitute of sensors, displays, communication protocolls etc.
It has a built-in command shell and web browser, so you can change all configuration by logging into the console or sending MQTT commands. It has compile-time switches so you can built-in only the code for the sensors, displays or whatever which you need.
So if you want to implement an iot device for some sensor, you can start with tasmota, and you do not have to care about network, mqtt, formware update configuration, encryption, certificates, command interpreter, etc. and just focus on your devices. It has a builtin scheduler and rule system.
Many of the features and sensors of iSpindel would be covered from scratch already. Others would have to be added/merged.
But then you could e.g. change the polling intervall or the calibration without even opening the device, because everything would show up in the embedded web browser.
